Prometheus app metrics

mjholder opened the pull request:
RHCLOUD-27159

mjholder committed:
added prometheus dependency

mjholder committed:
prometheus runs on same port as django

mjholder committed:
remove noreload flag

Victoremepunto commented on Makefile:
```diff
@@ -18,3 +18,15 @@ ifndef VIRTUAL_ENV
 	$(warning VIRTUAL_ENV variable present, already within a virtual environment?)
 endif
+
+cookie-teardown:
```
I think this requires the `venv_check` as a prerequisite to ensure we are on a virtual environment (so that `deactivate` is present in the environment)

mjholder replied:
Good point! I added the check now

Victoremepunto commented on {{cookiecutter.project_name}}/src/{{cookiecutter.project_name}}/settings.py:
```diff
@@ -37,16 +37,19 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'rest_framework',
+    'django_prometheus',
```
shouldn't this be `django-prometheus` ? I presume we're using https://pypi.org/project/django-prometheus/

mjholder replied:
That is correct, if you look under the quickstart section for the docs you linked. It's strange that there is an underscore rather than a dash.

Victoremepunto reviewed (changes requested):
I like the simplicity of this setup ! please, can you check my comment on the package name (I think there's an alias and that's why it works but I believe we should name it properly in the dependencies list.

I would like to see some section added to the README please, letting the user know what is configured regarding to exporting metrics via Prometheus (something really simple, with a link to the library for further info).

mjholder committed:
check for venv before deactivate

Signed-off-by: Matt Holder <mjholder96@gmail.com>

mjholder committed:
put venv_check in correct place

Signed-off-by: Matt Holder <mjholder96@gmail.com>

mjholder committed:
removed cookie cutter targets

Signed-off-by: Matt Holder <mjholder96@gmail.com>

mjholder committed:
Added info on how to use prometheus metrcs

Signed-off-by: Matt Holder <mjholder96@gmail.com>

mjholder commented:
> I like the simplicity of this setup ! please, can you check my comment on the package name (I think there's an alias and that's why it works but I believe we should name it properly in the dependencies list.
> 
> I would like to see some section added to the README please, letting the user know what is configured regarding to exporting metrics via Prometheus (something really simple, with a link to the library for further info).

I updated the docs on how to use the prometheus_client. I also removed the attempt at automation for cookiecutter. It wasn't working probably and I figured it would be better as it's own PR once I get it working.

mjholder committed:
Merge branch 'main' into prometheus-app-metrics

Commit:
add middleware to track total requests with parity to the go app

Commit:
Merge branch 'prometheus-app-metrics' of https://github.com/mjholder/cookiecutter-backend-starter-app-python into prometheus-app-metrics

Commit:
fix eof

mjholder committed:
linting

mjholder committed:
linting

mjholder committed:
linting

Victoremepunto committed:
Merge branch 'main' into prometheus-app-metrics

Signed-off-by: Victor M. <vmugicag@redhat.com>

mjholder committed:
Merge branch 'main' into prometheus-app-metrics
